modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/transformers/ActionsComponentPropsTransformer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {render} from 'frontend-js-react-web';
+import React from 'react';
+
+import SidebarPanel from '../SidebarPanel';
+
+export default function propsTransformer({
+	items,
+	sidebarContainerSelector,
+	...otherProps
+}) {
+	const sidebarRefPanel = React.createRef();
+
+	const actions = {
+		showInfo() {
+			showSidebar();
```

Comment:
This function isn't defined yet is it?

Reply:
Which one? `showSidebar`? https://github.com/liferay-frontend/liferay-portal/pull/142/files#diff-f5b7086370c04e63010e86d6871bfebbR44

The idea for the next pr is that `showSidebar` receives a `fetchURL` and a View, so the code in the end will be:

```
const actions = {
		showInfo(fetchURL) {
			showSidebar(fetchURL, SidebarPanelInfoView);
		},
		showMetrics(fetchURL) {
			showSidebar(fetchURL, SidebarPanelMetricsView);
		},
	};

...

const showSidebar = (fetchURL, View) => {
		const sidebarPanel = _getSidebarPanel();

		if (!sidebarPanel) {
			render(
				SidebarPanel,
				{
					View,
					fetchURL,
					onClose: hideSidebar,
					ref: _setSidebarPanel,
				},
				document.querySelector(sidebarContainerSelector)
			);
		}
		else {
			sidebarPanel.open(fetchURL, View);
		}
	};
```

Reply:
That caught my attention for a moment as well, but I figured hoisting is a feature, right?

Reply:
When any action is called `showSidebar` should be already defined, so yep, hoisting is a feature here :P